### PR TITLE
Fixing Issue #21718: changing logic of implicit bean discovery according to CDI 2.0 spec.

### PR DESCRIPTION
**Pull request by ankathur**
According to CDI 2.0 spec section 12.1 "Archives with extensions and no beans.xml file are not candidates for implicit bean discovery"

#21718

### CODE REVIEW

**Comment by ankathur** [comment]
@glassfishrobot Run CI Tests please

**Comment by arindam-bandyopadhyay** [comment]
@ankathur Please consider putting more descriptive title of the PR

**Review by jjsnyder — DISMISSED** [review]
Looks good.

File: appserver/web/weld-integration/src/main/java/org/glassfish/weld/BeanDeploymentArchiveImpl.java [hunk 1]
```diff
@@ -473,8 +473,8 @@ private void populate(Collection<com.sun.enterprise.deployment.EjbDescriptor> ej
 
                                 if (!bdMode.equals(BeanDiscoveryMode.ANNOTATED) || isImplicitBeanArchive(context, weblibJarArchive)) {
                                     weblibJarsThatAreBeanArchives.add(weblibJarArchive);
-                                }
-                            }
+                                }
```
Comment (arindam-bandyopadhyay) [inline appserver/web/weld-integration/src/main/java/org/glassfish/weld/BeanDeploymentArchiveImpl.java]: Consider fixing the formatting

**Review by arindam-bandyopadhyay — CHANGES_REQUESTED** [review]

**Comment by ankathur** [comment]
@glassfishrobot Run CI Tests please

File: appserver/web/weld-integration/src/main/java/org/glassfish/weld/BeanDeploymentArchiveImpl.java [hunk 2]
```diff
@@ -470,7 +470,7 @@ private void populate(Collection<com.sun.enterprise.deployment.EjbDescriptor> ej
                                                CDILoggerInfo.WEB_INF_LIB_CONSIDERING_BEAN_ARCHIVE,
                                                new Object[]{entry});
                                 }
-
+
```
Comment (arindam-bandyopadhyay) [inline appserver/web/weld-integration/src/main/java/org/glassfish/weld/BeanDeploymentArchiveImpl.java]: do you really need to change this file ?

File: appserver/web/weld-integration/src/main/java/org/glassfish/weld/BeanDeploymentArchiveImpl.java (hunk 1 above)
Comment (ankathur) [inline appserver/web/weld-integration/src/main/java/org/glassfish/weld/BeanDeploymentArchiveImpl.java]: updated the file.

**Review by ankathur — COMMENTED** [review]
removed BeanDeploymentArchiveImpl from PR

**Review by arindam-bandyopadhyay — APPROVED** [review]

**Comment by arindam-bandyopadhyay** [comment]
@glassfishrobot Run CI Tests please

**Pull request merged**